src/canopy/knowledge_base/knowledge_base.py

```diff
@@ -259,7 +262,10 @@ def verify_index_connection(self) -> None:
             ) from e
 
     def create_canopy_index(self,
-                            indexed_fields: Optional[List[str]] = None,
+                            spec: Union[Dict, ServerlessSpec, PodSpec] = ServerlessSpec(
+                                cloud='aws',
+                                region='us-west-2'
+                            ),
                             dimension: Optional[int] = None,
                             index_params: Optional[dict] = None
```

Comment:
I think this was replaced by `spec`, right? (for example, `pod_type` used to go in here, but now it will be part of `spec` in case of non-serverless index).  
We can probably delete this argument.

Reply:
still timeout and metric can be in index_params

Reply:
So let's just split them to `metric: str = 'cosine', timeout: Optional[int]...` and that's it.  
I don't see a point keeping this arbitrary `index_params` dict which is different from the `spec`, it's very confusing.